### PR TITLE
feat: allow to upload firmware in zip files

### PR DESCRIPTION
**Pull request by robertsLando**
Fixes https://github.com/zwave-js/zwave-js-ui/issues/3983

### CODE REVIEW

**Comment by robertsLando** [comment]
@AlCalzone About `tryUnzipFirmwareFile` should I check the file extension first and then use it before guess ?

**Comment by AlCalzone** [comment]
Almost correct. If ".zip" use `tryUnzipFirmwareFile`, otherwise use `guessFirmwareFileFormat`.
`tryUnzipFirmwareFile` does the guessing aswell, otherwise the zip file would have to be parsed twice.

**Comment by robertsLando** [comment]
Ok so the code I wrote should be correct, right?

**Review by AlCalzone — APPROVED** [review]

**Comment by AlCalzone** [comment]
Looks good. Just need to wait for 14.1.0 to finish building.

File: api/lib/ZwaveClient.ts [hunk 1]
```diff
@@ -3892,13 +3894,28 @@ class ZwaveClient extends TypedEventEmitter<ZwaveClientEventCallbacks> {
 			const firmwares: Firmware[] = []
 
 			for (const f of files) {
-				const { data, name, target } = f
+				let { data, name } = f
 				if (data instanceof Buffer) {
```
Comment (robertsLando) [inline api/lib/ZwaveClient.ts:3898]: @AlCalzone What's the correct way to check buffer here? Should I check if it's an Uint8Array

Comment (AlCalzone) [reply api/lib/ZwaveClient.ts:3898]: To test whether something is a `Uint8Array`, use the `isUint8Array` function exported from `node:util/types` (not portable) or `@zwave-js/shared` (portable).

**Pull request merged**